tests/ui/lifetimes/issue-26638.stderr

```diff
@@ -44,6 +44,10 @@ LL | fn parse_type(iter: Box<dyn Iterator<Item=&str>+'static>) -> &str { iter.ne
    |
    = note: expected reference `&str`
                    found enum `Option<&str>`
+help: you may missing `unwrap` or `expect(...)` here
```

Comment:
I haven't checked if the necessary information is available in `demand.rs` but we could suggest inserting the try operator `?` instead of `.unwrap()` or `.expect(…)` when the return type *expectation* implements `Try` like `Option<_>` or `Result<_>` :thinking:

We do have some diagnostics suggesting `?` already.